src/linear_algebra/finsupp.lean

```diff
@@ -476,7 +476,24 @@ begin
   apply total_emb_domain R ⟨f, hf⟩ l
 end
 
-theorem span_eq_map_total (s : set α):
+theorem span_eq_map_total (s : set M) :
+  span R s = submodule.map (finsupp.total s M R coe) ⊤ :=
```

Comment:
Isn't the preferred spelling of `submodule.map (finsupp.total s M R coe) ⊤` `(finsupp.total s M R coe).range`?

Reply:
Which also has a very short proof:
```lean
theorem span_eq_range_total (s : set M) :
  span R s = (finsupp.total s M R coe).range :=
by rw [range_total, subtype.range_coe_subtype, set.set_of_mem_eq]
```